src/main/java/com/celements/appScript/AppScriptService.java

```diff
@@ -169,26 +271,32 @@ public String getScriptNameFromURL() {
   public boolean isAppScriptRequest() {
     // TODO exclude isOverlayRequest
     return isAppScriptXpageRequest() || isAppScriptActionRequest() || isAppScriptSpaceRequest()
-        || isAppScriptOverwriteDocRef(getContext().getDoc().getDocumentReference());
+        || isAppScriptOverwriteDocRef(mContext.getDocRef().orElse(null));
   }
 
   @Override
   public boolean isAppScriptOverwriteDocRef(DocumentReference docRef) {
-    String overwriteAppDocs = getContext().getWiki().getXWikiPreference(
-        APP_SCRIPT_XWPREF_OVERW_DOCS, APP_SCRIPT_CONF_OVERW_DOCS, "-", getContext());
-    List<DocumentReference> overwAppDocList = new ArrayList<>();
-    if (!"-".equals(overwriteAppDocs)) {
-      for (String overwAppDocFN : overwriteAppDocs.split("[, ]")) {
-        try {
-          DocumentReference overwAppDocRef = webUtils.resolveDocumentReference(overwAppDocFN);
-          overwAppDocList.add(overwAppDocRef);
-        } catch (Exception exp) {
-          LOGGER.warn("Failed to parse appScript overwrite docs config part [" + overwAppDocFN
-              + "] of complete config [" + overwriteAppDocs + "].");
+    try {
+      String overwriteAppDocs = wikiProvider.await(Duration.ofSeconds(60)).getXWikiPreference(
```

Comment:
it should be safe to use `wikiProvider.get()` (which fast fails) since request initialisation already awaits the wikiProvider.